lib/routes/charge/items/catalog_item.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:breez/bloc/user_profile/currency.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/flushbar.dart';
+import 'package:fixnum/fixnum.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_slidable/flutter_slidable.dart';
+
+import 'item_avatar.dart';
+
+class CatalogItem extends StatelessWidget {
+  final AccountModel accountModel;
+  final PosCatalogBloc posCatalogBloc;
+  final Item _itemInfo;
+  final bool _lastItem;
+  final Function(AccountModel accountModel, String symbol, Int64 itemPriceInSat)
+      _addItem;
+
+  CatalogItem(this.accountModel, this.posCatalogBloc, this._itemInfo,
+      this._lastItem, this._addItem);
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(alignment: Alignment.bottomCenter, children: <Widget>[
+      Slidable(
+        actionPane: SlidableDrawerActionPane(),
+        actionExtentRatio: 0.25,
+        secondaryActions: <Widget>[
+          IconSlideAction(
+            caption: 'Delete',
+            color: Colors.red,
+            icon: Icons.delete_forever,
+            onTap: () {
+              DeleteItem deleteItem = DeleteItem(_itemInfo.id);
+              posCatalogBloc.actionsSink.add(deleteItem);
+              deleteItem.future.then((_) => showFlushbar(context,
+                  message: "${_itemInfo.name} is successfully deleted"));
+            },
+          ),
+        ],
+        key: Key(_itemInfo.id.toString()),
+        child: ListTile(
+          leading: _buildCatalogItemAvatar(),
+          title: Text(
+            _itemInfo.name,
+            style: theme.transactionTitleStyle,
+            overflow: TextOverflow.ellipsis,
+          ),
+          trailing: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            crossAxisAlignment: CrossAxisAlignment.end,
+            children: <Widget>[
+              Row(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  mainAxisSize: MainAxisSize.min,
+                  children: <Widget>[
+                    Text(
+                      _formattedPrice(),
+                      style: theme.transactionAmountStyle,
+                    )
+                  ]),
+            ],
+          ),
+          onTap: () {
+            _addItem(accountModel, _itemInfo.currency, _itemPriceInSat());
+          },
+        ),
+      ),
+      Divider(
+        height: 0.0,
+        color: _lastItem
+            ? Color.fromRGBO(255, 255, 255, 0.0)
+            : Color.fromRGBO(255, 255, 255, 0.12),
+        indent: 72.0,
+      ),
+    ]);
+  }
+
+  String _formattedPrice({bool userInput = false, bool includeSymbol = true}) {
+    return Currency.fromSymbol(_itemInfo.currency) != null
+        ? Currency.fromSymbol(_itemInfo.currency).format(
+            Int64(_itemInfo.price.toInt()),
+            userInput: userInput,
+            includeSymbol: includeSymbol)
+        : accountModel.fiatConversionList
```

Comment:
I think it makes sense to add `formatFiat` function on `FiatCurrency` instead of this round trip.

Reply:
[Added formatFiat function to FiatConversion](https://github.com/breez/breezmobile/pull/289/commits/cd4ace7f2239fbc2ab97a5b1b00f3dafaccc55c1)
[Replaced with new methods](https://github.com/breez/breezmobile/pull/289/commits/00d633e7be17973398773a5034ca234b5c3d7b05)